api/app/src/command/medical/document/document-query.ts

```diff
@@ -110,20 +110,22 @@ export const updateDocQuery = async ({
     });
 
     if (existing) {
-      const convertSuccess = existing.data.documentQueryProgress?.convert?.successful
-        ? existing.data.documentQueryProgress?.convert?.successful + 1
+      const docQueryProgressConvert = existing.data.documentQueryProgress?.convert;
+
+      const convertSuccess = docQueryProgressConvert?.successful
+        ? docQueryProgressConvert?.successful + 1
         : 1;
 
-      const convertError = existing.data.documentQueryProgress?.convert?.errors
-        ? existing.data.documentQueryProgress?.convert?.errors + 1
+      const convertError = docQueryProgressConvert?.errors
+        ? docQueryProgressConvert?.errors + 1
         : 1;
```

Comment:
Technically we are doing that. I set the variables on the top to avoid cluttering even further the update function but we wont actually insert the increment. I can update the logic though if thats confusing

```
  convertResult === "success"
    ? {
        ...existing.data.documentQueryProgress?.convert,
        status: docQueryProgressStatus,
        successful: convertSuccess,
      }
    : {
        ...existing.data.documentQueryProgress?.convert,
        status: docQueryProgressStatus,
        errors: convertError,
      },
```

Reply:
Now I see we're checking the status at the actual update like you mentioned, but we're still using both incremented const `convertSuccess` and `convertError` to calculate `docQueryProgressStatus` - only one of those should be increased to calculate the progress status, right?

Reply:
Mmm good point let me fix

Reply:
I suggest we move the check `convertResult === "success"` to the top of the `if` and use it to increment one OR the other.

Reply:
Refactored the logic